src/shadowbox/server/libev_shadowsocks_server.ts

```diff
@@ -93,17 +96,36 @@ export class LibevShadowsocksServer implements ShadowsocksServer {
     });
   }
 
-  public startInstance(id: string, portNumber: number, password: string, encryptionMethod):
-      Promise<ShadowsocksInstance> {
-    logging.info(`Starting server on port ${portNumber}`);
-    this.portId[portNumber] = id;
+  update(newKeys: AccessKey[]): Promise<void> {
```

Comment:
This function could really use a comment - looks like it starts new keys before stopping old ones?